### ISSUE TITLE
Failed to open file Bam file

### ISSUE THREAD

**Issue by Admera2020** [DaehwanKimLab/hisat-genotype#36]
Hello, 

I am running hisat-genotype with my samples, below is my full command 

`hisatgenotype --base hla --keep-alignment --locus-list A -z /fast-data/BI/Nihir/miniconda3/custom/hisatgenotype/indicies --out-dir hisatgenotype_A1 -1 SampleA1_R1.fastq.gz -2 SampleA1_R2.fastq.gz `

After it finishes extracting reads I keep getting an error 
` Extracting reads from SampleA1_R1.fastq.gz
[E::hts_open_format] Failed to open file "SampleA1_R1_fastq_gz-hla-extracted-1_fq.bam" : No such file or directory
samtools index: failed to open "SampleA1_R1_fastq_gz-hla-extracted-1_fq.bam": No such file or directory
         A
Error in running HISATgenotype: Incomplete Align Return` 
The report file in the output directory shows the correct version of hisat2 and hisat-genotype 

```

VERSIONS:
HISAT2 - 2.2.1

HISAT-genotype - 1.3.2
Database - Database /fast-data/BI/Nihir/miniconda3/custom/hisatgenotype/indicies/hla derived from HISATgenotype DB version: NONE
 COMMAND:
/fast-data/BI/Nihir/miniconda3/bin/hisatgenotype --base hla --keep-alignment --locus-list A -z /fast-data/BI/Nihir/miniconda3/custom/hisatgenotype/indicies --out-dir hisatgenotype_A1 -1 SampleA1_R1.fastq.gz -2 SampleA1_R2.fastq.gz

                hisat2 graph

```

Thanks in advance for your help.

Best, 
Nihir.

**Comment by chbe-helix** [comment]
Hi Nihir,

It looks like HISAT-genotype is failing at the alignment step. Can you align your files with HISAT with the following command and get alignments?

```
hisat2 -x genotype_genome --no-spliced-alignment --max-altstried 64 -X 1000 -1 SampleA1_R1.fastq.gz -2 SampleA1_R2.fastq.gz
```

(Note: set appropriate file paths accordingly)

Thanks,
Chris

**Comment by Admera2020** [comment]
Hello Chris, 

Sorry, I am not sure if I am running it correctly but still getting error. 

Step1: Extracted reads and it failed after the extraction step
```
DIR="/fast-data/BI/Nihir/miniconda3/custom/hisatgenotype/indicies"
hisatgenotype --base hla --keep-alignment --locus-list A -z $DIR --out-dir hisatgenotype_A1 -1 SampleA1_R1.fastq.gz -2 SampleA1_R2.fastq.gz

```

Step2 : Perform mapping and sort the bam file 

```
hisat2 -x $DIR/genotype_genome --no-spliced-alignment --max-altstried 64 -X 1000 -1 SampleA1_R1.fastq.gz -2 SampleA1_R1.fastq.gz | samtools view -bh - > O-SampleA1.bam
samtools sort SampleA1.bam  > SampleA1_sort.bam
samtools index SampleA1_sort.bam 

```

Step3 re-run hisatgenotype

`hisatgenotype --base hla --keep-alignment --locus-list A -z $DIR --out-dir hisatgenotype_A1 -1 SampleA1_R1.fastq.gz -2 SampleA1_R2.fastq.gz --pass-extract --bamfile SampleA1_sort.bam >  SampleA1_res.txt
`

 I see the below error as soon as I ran Step3
`Error in running HISATgenotype: Incomplete Align Return`

**Comment by chbe-helix** [comment]
Hi Nihir,

The error you are seeing occurs when there are no lines returned from the alignment step with hisat2. What this is telling me is that your bam file may be empty or have some error that prevents it from being properly read. Are their aligned reads in your bam file?

Thanks,
Chris

**Comment by Admera2020** [comment]
Hello Chris, 

Yes there are definitely mapped reads, below is the stats from samtools flagstats

```
62266589 + 0 in total (QC-passed reads + QC-failed reads)
4620301 + 0 secondary
0 + 0 supplementary
0 + 0 duplicates
56742924 + 0 mapped (91.13% : N/A)
57646288 + 0 paired in sequencing
28823144 + 0 read1
28823144 + 0 read2
1050 + 0 properly paired (0.00% : N/A)
52119066 + 0 with itself and mate mapped
3557 + 0 singletons (0.01% : N/A)
410746 + 0 with mate mapped to a different chr
6424 + 0 with mate mapped to a different chr (mapQ>=5)

```
Nihir

**Comment by chbe-helix** [comment]
Hi Nihir,

Great! That answers that possibility. The next possibility is there is a problem with reading the BAM file. Could you tell me what the output of the following command is?:

```
samtools view SampleA1_sort.bam | sort -k 1,1 -s | head -20
```

Thanks,
Chris

**Comment by Admera2020** [comment]
Hello Chris,

Here is the output
```
NS500466:676:HJVYJAFX2:1:11101:10030:16484      163     6       33016887        60      151M    =       33016959        218     ACTAGGACTGGGATTAAGGGACGTTCCCCCTTTGTAGCCATCTGTGGGCAGGGGATGCTCTGGGGTATCCACTGGGGCAGGAGAGGAGGGAAACAGAGGGAGAGGAGACTGGGGAGGGAGTGGGGACGCCAGGAGCTCCTATATTTGACTG        AAAAAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAEEAEEEEEEEEEEEEEEEEEEAEEEEEEEEEEEEEEEEEEEEEEEEAEAAAEEEEAEEEA/EAAEEAAEAAE        AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:151        YS:i:-20        YT:Z:CP NH:i:1
NS500466:676:HJVYJAFX2:1:11101:10030:16484      83      6       33016959        60      5S146M  =       33016887        -218    CCCAGTGGGGCTGGAGAGGAGGCAAACTGTGGGAGAGGAGACTGGGGAGGGAGGGGGGACGCCAGGAGCTCCTATATTTGACTGGTCCCTGGGCGGGAGTCCGGGTGAGAGGTGTCATTCCTCAAGGAGAGGGGTGCCAAAGGGGTCTGGG        /A////EEE////////</////E<E////EA////E//////EE<//EE//E/EEE/EEAEA<EEEEEEE/E<EEEEEEEEEEEAEEEEEEEEAEEEEEEAAAEEEEEEEEEEAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAAAAAA        AS:i:-20        XN:i:0  XM:i:5  XO:i:0  XG:i:0  NM:i:5  MD:Z:6A10G4A1A23T97     YS:i:0  YT:Z:CP NH:i:1
NS500466:676:HJVYJAFX2:1:11101:10033:13966      99      6       29829529        60      151M    =       29829632        252     CTGGCAGCGGGATGGGGAGGACCAGACCCAGGACGTGGAGCTCGTGGAGACCAGGCCTGCAGGGGATGGAACCTTCCAGAAGTGGGCAGCTGTGGTGGTGCCTTCTGGAGAGGAGCAGAGATACACGTGCCATGTGCAGCATGAGGGGCTG        AAAAAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAEEEEEEEEEEEEEE<EEEEEEEEEEEEEEEEEEEEEEAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAEEEEEEEEEEEEEEEEEEAAEEE<AEE<EEEEEEAAAEEE        AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:151        YS:i:0  YT:Z:CP NH:i:1
NS500466:676:HJVYJAFX2:1:11101:10033:13966      147     6       29829632        60      149M    =       29829529        -252    TCTGGAGAGGAGCAGAGATACACGTGCCATGTGCAGCATGAGGGGCTGCCAGAGCCCCTCATGCTGAGATGGAGTAAGGAGGGAGATGGAGGCATCATGTCTGTTAGGGAAAGCAGGAGCCTCTCTGAAGACCTTTAACAGGGTCGGTG  /A<EAAAAEA<EEEEEEEA<EEEEEEEEEEEEEEEAEEEEEEEAEEEE<EEEEEEEEE<A/<AEAAE<EEEAEEEEEEEEEEEEEEEEEEEEEEEAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAEEEEEEEEAAAAA  AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:50G98      YS:i:0  YT:Z:CP NH:i:1  Zs:Z:50|S|hv1039
NS500466:676:HJVYJAFX2:1:11101:10060:5241       163     6       32944448        60      151M    =       32944463        165     ATCATGCCCTTGCCTAGATCCTCATGGAAGCTCACCCATATAATAATCAAGATTAGTTGAACCCAACACTGACCCCTCTAACCCGCACCCCTACCAAAGGGCAAGTAGGGAAACAGACCAACAGAGATGTTACCTTCTGAATAATTGGACC        AAAAAEEAEEEEEEEEEEEEEEAEEEEEEEEEEEE<EEEEEEEEEEEAAEEEEEEEEAAEAA6AEEEEEAEEEE<EEEEEEEAEAEEEAAAE<AEEEEE<EA<E<EEEEEEEAEEEE</A/E/<AAEEA<AEE/EE<E<EEA<EA<<<A/A        AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:151        YS:i:0  YT:Z:CP NH:i:1
NS500466:676:HJVYJAFX2:1:11101:10060:5241       83      6       32944463        60      150M    =       32944448        -165    AGATCCTCATGGAAGCTCACCCATATAATAATCAAGATTAGTTGAACCCAACACTGACCCCTCTAACCCGCACCCCTACCAAAGGGCAAGTAGGGAAACAGACCAACAGAGATGTTACCTTCTGAATAATTGGACCCAGGAAGAGGAGTG EEEEAEEEEE/AAAA<E<E<A<E<E/E<AEEEAEEAEEAAA/EEEEE/EEEAAEAEEEEEEEEAEEEEEE/EE/EE/EEEEEEE6E<AEEEE<EEAEEAEEEEEEEEE/EAEEEEEE/EEEEEEEEEEEEEAAEEEAEEEEEEEEAAAAA AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:150        YS:i:0  YT:Z:CP NH:i:1
NS500466:676:HJVYJAFX2:1:11101:10077:3596       163     6       29927355        60      151M    =       29927466        262     AGCGCAAGTGGGAGGCAGAAGAATTTGCAGAGCAGATCAGGGCCTACCTGGAGGGCACGTGCGTGGAGAGGCTCGCAGACACCTGGAGAACGGGAAGGAGACGCTGCAGCTCACGGGTACCAGGGAACACAAGACGTCTCCCTGATCGCCT        AAAAAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAEEEEEEEAEEEEEEEEEEEAEEEEEEEEEEEEEEEEEEEE<EEEEEEEEEEEEEEEAEEEEEAAEAEEEA        AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:151        YS:i:-6 YT:Z:CP NH:i:1
NS500466:676:HJVYJAFX2:1:11101:10077:3596       83      6       29927466        60      151M    =       29927355        -262    CACGGGAACCAGGCAACACAAGACGTCTCCCTGATCGCCTGTAGATCTCCTGGGCTGGCTTCCCACAAAGAGAGAAGGAAAATGGGACCAACACTAGAATGTCGTCCTCTCTCTGGTCCTGAGGGAGAGGAATCCTCCTGGGTTTCCAGAT        6//AE</EEAA<A/E/A/<A</EEEEE<EEEAAEAEEEEEEEEEEAEEEEEEAEEAEEEA<EEEEEEEEEEEEEEAAEEEEEEEAAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAAAAA        AS:i:-6 ZS:i:-14        XN:i:0  XM:i:2  XO:i:0  XG:i:0  NM:i:2  MD:Z:6T6G137    YS:i:0  YT:Z:CP NH:i:1
NS500466:676:HJVYJAFX2:1:11101:10079:1595       163     6       29791816        60      151M    =       29791886        220     ACGGGGATTCCGTGATGCAAAATGAAGAGAGAGGAACGGGGCCCATTTCGAGGGTTTCTCCCTGGTTTCTCAGACAGCTCCTGGGCCAAGACTCAGGGAAACGTTGAGACAGAGCGCTTGGCACAGAAGTAGCGGGGTCAGGGCGAAGTCC        AAAAAEEEEEEEEEEEEEEAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE/EEEEEEEEEEA<AEAAEAEAAEEEEEEEAEAEAAAAEEEEEAEEEEEEEE        AS:i:-5 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:21G129     YS:i:-5 YT:Z:CP NH:i:1
NS500466:676:HJVYJAFX2:1:11101:10079:1595       83      6       29791886        60      150M    =       29791816        -220    CAGACAGCTCCTGGGCCAAGACTCAGGGAAACGTTGAGACAGAGCGCTTGGCACAGAAGTAGCGGGGTCAGGGCGAAGTCCCAGGGCCTCAGGCATGGCTCTCAGGATCTCAGGGCCCAAAGGCGGTGTATGGATTGGGGAGGCCCAGCG EEEEEEEAEEEEAAAAAEEEEEEA<AAEEEEEEEEEEEEAEEEEEEEAEAAEAEEEEEEEAEEEEEEEEAEAAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAAAAA AS:i:-5 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:114C35     YS:i:-5 YT:Z:CP NH:i:1
NS500466:676:HJVYJAFX2:1:11101:10082:19857      163     6       32670442        60      151M    =       32670492        201     AGAGGTGCTCTAGTCTCCTGTGATTCCCAGCTCAGCAGTGACATCAGGGATAAGAGATGGGAAGGGATGGGTCAGAAGGAGCTCTGTCCTTTGTCTTGTGGGGCCCACAGTAACAGAAACTCAATATCCCCTTACGCCACTCCACGGTGAT        AAAAAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAEEEEEAEAAA<AEEEEAAAAEEEEEEEEEEEE        AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:35T29A20C0T19T43   YS:i:-24        YT:Z:CP NH:i:1  Zs:Z:35|S|hv12559,29|S|hv12560,20|S|hv12564,0|S|hv12565,19|S|hv12567
NS500466:676:HJVYJAFX2:1:11101:10082:19857      83      6       32670492        60      151M    =       32670442        -201    AAAGATATGGGATGGGTTGGGTCAGAAGGAGCTCTGTCCTTTGTCTTGTGGGGCCCACAGTAACAGAAACTAAATATCCCATAACGCCACTCCACGGTGATGGGGCTCTGGAGGCTGGGGTGCTCCACGTGGCAGGTGTAGACGTCTCCAC        /6//////A</A/<////AA6////A/E//<////A/</EEE<E/E/E/EEE/EE/E////EA///EE////AA/E/EEE/E//E<E/EE/EE/AEA//<AEEEEEAEAEE/EEAAEEEEEEEAEEEEE/EEEAEEAEAEEEEEEEAAAAA        AS:i:-24        ZS:i:-24        XN:i:0  XM:i:8  XO:i:0  XG:i:0  NM:i:8  MD:Z:0T4G6A2A0A19C0T19T13C8C1T68        YS:i:0  YT:Z:CP NH:i:1  Zs:Z:36|S|hv12564,0|S|hv12565,19|S|hv12567
NS500466:676:HJVYJAFX2:1:11101:10095:14447      163     6       32651157        60      151M    =       32651222        214     ACCTCAACGACTTCATATCCACAGAGCCTGAGATTCCAGCCCCTATGTCAGAGCTCACAGAGACTGTGGTCTGCGCCCTGGGATTGTCTGTGGGCCTCGTGGGCATTGTGGTGGGCACTGTCTTCATCATCCGAGGCCTGCGTTCAGTTGG        AAAAAEEEEEEEEEEEEEEEEEEEEEEEEEAEEEEEEEEEA<EEEEEEEAEEAEEEEEEEEEEEEE/EEEEAE<EEAEEE<EEE<EEEEEEEEEEEEEEEEEEEEEAE<A<<EA/EAEEAEAEEEEEAAEEEEEEEEEAAAEAEA/EE/<<        AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:8A6C0T65G49A18     YS:i:0  YT:Z:CP NH:i:1  Zs:Z:8|S|hv12106,6|S|hv12107,0|S|hv12108,65|S|hv12115,49|S|hv12121
NS500466:676:HJVYJAFX2:1:11101:10095:14447      83      6       32651222        60      149M    =       32651157        -214    GTGGTCTGCGCCCTGGGATTGTCTGTGGGCCTCGTGGGCATTGTGGTGGGCACTGTCTTCATCATCCGAGGCCTGCGTTCAGTTGGTGCTTCCAGACACCAAGGGCCCTTGTGAATCCCATCCTGGAATGGAAGGTAAGATTGAGATTT  EEEEEEEEEA<6AA//EEEAEEEEEEAA<AAEEAA<AA</EAEEE/EE<<EEEEEEE/EEEEEEEE/EEEAEE<<E<E/EEEEEEE/EEAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE/EEAEEAEEEEEEEEEEEEEAAAAA  AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:17G49A60G17C1G0    YS:i:0  YT:Z:CP NH:i:1  Zs:Z:17|S|hv12115,49|S|hv12121,60|S|hv12126,17|S|hv12129,1|S|hv12130
NS500466:676:HJVYJAFX2:1:11101:10097:4632       163     6       32962124        60      151M    =       32962154        180     CTTTTCCTCACAAAGCTGAGTGGGCTCCCTAGGTTCAGGATGGAAGTAAATAGGAGTACCATCTTACCTTCAGGGACGGCCCAGGAGTGGGGTAGCAGCCACAGAAGTGGTAACATCTGTAGCAGCGCAGCTCCTTGGTTCTGTTCATGAC        AAAAAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAEEAEEEEEEAEEEEEEEAEEE6AEEAEE<EAEEEEEEEEEAEEAEEE<EEEEEEEEEA<<AA<<EEAAAEAEEAAAAAAEEAAAEAEEEEE/        AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:151        YS:i:0  YT:Z:CP NH:i:1
NS500466:676:HJVYJAFX2:1:11101:10097:4632       83      6       32962154        60      150M    =       32962124        -180    AGGTTCAGGATGGAAGTAAATAGGAGTACCATCTTACCTTCAGGGACGGCCCAGGAGTGGGGTAGCAGCCACAGAAGTGGTAACATCTGTAGCAGCGCAGCTCCTTGGTTCTGTTCATGACCCATACCTTCTTGCCACACAGTAGGTAGG E/EEEEEEEEAEEAAAEA<EEEAEEEEAEEEEEEEEEEAEEEEEEEEEEEEEEEEEEAEEEEEEEEEEEAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAAAAAA AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:150        YS:i:0  YT:Z:CP NH:i:1
NS500466:676:HJVYJAFX2:1:11101:10104:19808      163     6       29944230        60      151M    =       29944288        208     CACATATGACCCACCACCCCATCTCTGACCATGAGGCCACTCTGAGATGCTGGGCCCTGGGCTTCTACCCTGCGGAGATCACACTGACCTGGCAGCGGGATGGGGAGGACCAGACCCAGGACACGGAGCTTGTGGAGACCAGGCCTGCAGG        A6AAAEEEAEEEAEA/EEAEEEEEEEE/EEEAEEEEEEAEEE//EAEE/EEEEE<<EEA/<EAAE<EEEE<A/<EEAAEEEAEEEE<EAAE<EEEAE/<EAEE/AEE<E/EEE/AEEEEEEE/E/AEEEEA//AEAE/<<EE<</<6EEE<        AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:1G8T6G1T0G19C5G12A70C20    YS:i:0  YT:Z:CP NH:i:1  Zs:Z:1|S|hv2704,8|S|hv2709,6|S|hv2718,1|S|hv2721,0|S|hv2724,19|S|hv2735,5|S|hv2737,12|S|hv2740,70|S|hv2783
NS500466:676:HJVYJAFX2:1:11101:10104:19808      83      6       29944288        60      150M    =       29944230        -208    GGGCTTCTACCCTGCGGAGATCACACTGACCTGGCAGCGGGATGGGGAGGACCAGACCCAGGACACGGAGCTTGTGGAGACCAGGCCTGCAGGGGATGGAACCTTCCAGAAGTGGGCAGCTGTGGTGGTACCTTCTGGAGAGGAGCAGAG EEE6/<EEEEEAEE/</AA/EEEEAA<EEE</6<<EEAE<AAEEA/EE<EEEEAEEEEAEEEEEEEEEEEEEEEEEEEEEEEEEEEAAAAAEAEAEEAEEEEE6AEEEEEEA/EAEEEEEEEEAE/EE6EEEEAEE<AEEAEEEEAAAAA AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:1A70C44G11G9C10    YS:i:0  YT:Z:CP NH:i:1  Zs:Z:1|S|hv2740,70|S|hv2783,44|S|hv2819,11|S|hv2830,9|S|hv2836
NS500466:676:HJVYJAFX2:1:11101:10117:11388      163     6       30260372        60      150M    =       30260470        249     GGGTTGGTCGGGGCGGGAGCGGGGCTCGGTGTTCGGGGCTGACGGCGGGGGCGAGGCCATGGTTCTCACACCATCCAGAGGAAGCATGACTGCGACGTGGGCCCGACAGGCGCCTCCTCCGCAGGTATGAACAGTTCGCCTACGATGGCA //AAAE/EEEE/EEE6/EEEEAE/EAEAAEA/EEEEEAAEAAEEE/EEE//AEEE<EEAEE/E/EA</</EEE/E/6E/EEEE6EA/EEA//A<A<EAA/A/A/A<AA<<<A/AEEA/</A<6AA6<A/6EEAEE<E/<6A/EAE/A/E/ AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:88G61      YS:i:-4 YT:Z:CP NH:i:1  Zs:Z:88|S|hv3109
NS500466:676:HJVYJAFX2:1:11101:10117:11388      83      6       30260470        60      151M    =       30260372        -249    GGGCCCGACAGGCGCCTCCTCCGCAGGTATGAACAGTTCGCCTACGATGGCAAGGATTACATCGCCCTGAACGAGGACCTGCCCTCCTGGACCGCCGCGAACACAGCGGCTCAGATCTCCCAGCACAAGTGGGAAGCGGACAATTACTCAG        AE/AEEEA<</AEAEE/A<AAAE////AE<A/A/EAEEA6<EA<EA/A66/AEE/EAEA</AE////E/<<EAEE/EA/EAE/AAAA<////<<///E/EA/E/<E/6/AE6/A/EEAA6EE/EE</EEEE<E//E/EAEA6A6EEAAA/A        AS:i:-4 ZS:i:-8 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:82A60A7    YS:i:0  YT:Z:CP NH:i:1  Zs:Z:82|S|hv3110
```


Nihir

**Comment by chbe-helix** [comment]
Hi Nihir,

Perfect, okay, let's try the following command to get the script running using the bam file:

```
hisatgenotype --base hla --locus-list A -z $DIR --out-dir hisatgenotype_A1 --pass-extract --bamfile SampleA1_sort.bam
```
This should completely bypass extraction and we are looking for if it succeeds. The results from this may not be reliable so we'll have to see what comes out

Should that not work, lets try this script again:
```
hisatgenotype --base hla --keep-alignment --locus-list A -z $DIR --out-dir hisatgenotype_A1 -1 SampleA1_R1.fastq.gz -2 SampleA1_R2.fastq.gz
```
This last one is the same you ran before. What we will be looking for is the file __SampleA1_R1_fastq_gz-hla-extracted-1_fq.bam__ in the results folder created __hisatgenotype_A1__. 

The goal of this set of tests is to see if the read extraction contains a bug with your files. I'll also look at the output you posted a bit further to see if I can find a reason for the error directly depending on what this test shows. 

Sorry for the time it is taking to resolve you issue. Thanks for you patience through this process!

Thanks,
Chris

**Comment by Admera2020** [comment]
Hello Chris, 
	 

The first command existed with no errors and no output. The second common ran all the way. But I think I've pinpointed the issues here. 

So every command was giving me issues after the step of extraction so it seems that the problem has to be at switching to hisat2 step.
I went back and re-install the packages from scratch. Only this time I went through the manual route and it running fine now with the exact same data. 
What might have happened is this,  the auto-install `"bash setup.sh -r"` checks if the hisat2 is in PATH. For me I had hisat2 installed via conda, the script picked up the hisat2 and skip installing via the GitHub package. 
As per my understanding, I think the environment was having trouble linking hisatgenotyp to hisat2, so independently they were working fine but cause an error when I ran fully. 

Thanks for troubleshooting with me on this, appreciate your help. I believe it's safe to close the issue unless you think otherwise. 

Best, 
Nihir
